### DISCUSSION TITLE
Commanding from Open MCT

### DISCUSSION THREAD

**Discussion by melfvelin** [nasa/openmct#4113] [Community Q&A] [answered]
Hello, I'm struggling to find documentation about telecommand generation in Open MCT. Is this something that is normally done from Open MCT or would it be solved separately by each user?

**Comment by akhenry** [comment]
Hi! Thanks for the question. Commanding is not something that is addressed by Open MCT right now. Instead it is handled by whichever telemetry and commanding system Open MCT is paired with.

For example, [YAMCS](https://github.com/yamcs/yamcs) includes a comprehensive commanding capability, and a web interface for commanding that could be embedded in Open MCT layouts.

**Comment by akhenry** [comment, marked as answer]
Are you able to share any information about the project that you're working on? We may be able to provide some more specific guidance.

**Reply by melfvelin** [reply]
Thanks for the quick reply! I'm working on a student ground station that will the used to support a CubeSat using the GomSpace AX100 but also future CubeSat projects. Currently I'm looking at the suitability of Open MCT for the mission control center. For the commanding of the AX100 software from the supplier is used. However if more satellites are developed in the future it would be nice if operators could generate all commands from the same place. I'm not looking for very extensive functionality, we can develop the backend software ourselves, but rather how to generate commands from one place using Open MCT.

**Reply by jvigliotta** [reply]
@akhenry bumping!

**Reply by akhenry** [reply]
I see. Open MCT doesn't do commanding right now, but there are a couple of paths you could take here -

- [YAMCS](https://github.com/yamcs/yamcs) is a great telemetry and commanding system (I promise I don't work for them 😄). It provides a historical telemetry store, as well as a web interface for commanding. It is also out of the box compatible with Open MCT for displaying telemetry (but not commanding). Open MCT does allow you to build layouts with external web pages though, so you could incorporate YAMCS' commanding capability into Open MCT layouts.

- If you don't want to use YAMCS, and you can build a simple web GUI for commanding, you could include this GUI in Open MCT as an embedded web page.

**Reply by melfvelin** [reply]
Thanks for the answer, both alternatives sound like viable options.
